app/src/ui/suggested-actions/dropdown-suggested-action.tsx

```diff
@@ -0,0 +1,168 @@
+import * as React from 'react'
+import {
+  DropdownSelectButton,
+  IDropdownSelectButtonOption,
+} from '../dropdown-select-button'
+import { MenuIDs } from '../../models/menu-ids'
+import { executeMenuItemById } from '../main-process-proxy'
+import { sendNonFatalException } from '../../lib/helpers/non-fatal-exception'
+import classNames from 'classnames'
+
+export interface IDropdownSuggestedActionOption
+  extends IDropdownSelectButtonOption {
+  /**
+   * The title, or "header" text for a suggested
+   * action.
+   */
+  readonly title?: string
+
+  /**
+   * A text or set of elements used to present information
+   * to the user about how and where to access the action
+   * outside of the suggested action.
+   */
+  readonly discoverabilityContent?: string | JSX.Element
+
+  /**
+   * A callback which is invoked when the user clicks
+   * or activates the action using their keyboard.
+   */
+  readonly onClick?: (e: React.MouseEvent<HTMLButtonElement>) => void
+
+  /**
+   * Whether or not the action should be disabled. Disabling
+   * the action means that the button will no longer be
+   * clickable.
+   */
+  readonly disabled?: boolean
+
+  /**
+   * An image to illustrate what this component's action does
+   */
+  readonly image?: JSX.Element
+
+  /**
+   * The id of the menu item backing this action.
+   * When the action is invoked the menu item specified
+   * by this id will be executed.
+   */
+  readonly menuItemId?: MenuIDs
+}
+
+export interface IDropdownSuggestedActionProps {
+  /** The possible suggested next actions to select from
+   *
+   * This component assumes this is not an empty array.
+   */
+  readonly suggestedActions: ReadonlyArray<IDropdownSuggestedActionOption>
+
+  /** The value of the selected next action to initialize the component with */
+  readonly selectedActionValue?: string | undefined
+
+  readonly onSuggestedActionChanged: (action: string) => void
```

Comment:
Could this be generic? Maybe we can add a `<T extends string = string>` (I think that syntax is correct 😆 ) to `IDropdownSuggestedActionProps` and `IDropdownSelectButtonOption` and use it here like this:

```suggestion
  readonly onSuggestedActionChanged: (action: T) => void
```

That way we don't need to use a `string` and cast it to our enum when the callback is invoked.

Reply:
This would also affect `selectedActionValue`:
```typescript
readonly selectedActionValue?: T | undefined
```

Reply:
Nice suggestion! I had to type the component for this to work (or I didn't know how not too) but I think that makes sense anyways to say what kind of dropdown suggested next action it is.